tests/libs/estdlib/test_os.erl

```diff
@@ -0,0 +1,59 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2025 Jakub Gonet <jakub.gonet@swmansion.com>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(test_os).
+
+-export([test/0]).
+
+-include("etest.hrl").
+
+test() ->
+    ok = test_os_getenv(),
+    ok.
+
+test_os_getenv() ->
+    Machine = erlang:system_info(machine),
+    Platform =
+        case Machine of
+            "ATOM" -> atomvm:platform();
+            "BEAM" -> os:type()
+        end,
+    ok =
+        case Platform of
+            generic_unix -> validate_path(os:getenv("PATH"));
+            _ -> ok
```

Comment:
I missed this sorry. BEAM returns {unix, _} on unix, we could validate path in this case.

/cc @bettio

Reply:
Oops. I'll open a PR with `os:type`, so we can use it on both VMs